embassy-rp/src/uart/buffered.rs

```diff
@@ -421,9 +423,18 @@ impl<'d, T: Instance> BufferedUartTx<'d, T> {
         loop {
             let state = T::buffered_state();
             if state.tx_buf.is_empty() {
-                return Ok(());
+                break;
             }
         }
+        while self.busy() {}
+        Ok(())
+    }
+
+    /// Flush until the sender is no longer busy, using a yield-spinloop
+    pub async fn yielding_busyloop(&self) {
```

Comment:
every time you do `flush` you also want to do this. I think it'd make sense to make it part of `fn flush()` instead of remembering to call this at every `flush()` call.